### PR TITLE
spark built-in lineage

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
https://github.com/OpenLineage/OpenLineage/pull/2272

### CODE REVIEW

**Comment by pawel-big-lebowski** [comment]
TODO: describe interface for DataSourceV2

File: docs/integrations/spark/built_in_lineage.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+to expose dataset facets as well facets that relate to particular dataset only in the context of
+current run, like amount of bytes read from a certain dataset.
+
+There are two possible ways to identify dataset returned by a trait. The straightforward one
+is to identify it by `namespace` as `name` strings, which is a classical dataset identifier within
+OpenLineage convention.
+Oftentimes this approach can be cumbersome.
+There are many `command` nodes whose output dataset is `DatasourceV2Relation`, which is another
+node within the logical plan. In this case, it makes
+sense to delegate dataset extraction to other node, while still having the ability to enrich it
```
Comment (mobuchowski) [inline docs/integrations/spark/built_in_lineage.md:71]: I think it needs more context and explanation about this delegation process. It does not explain what happens when you use it - control goes back to OpenLineage integration and it uses the delegated LogicalPlan node to determine particular dataset. It also does not explain that you depend on that LogicalPlan node to have proper implementation whether using this mechanism or directly in OpenLineage.

File: docs/integrations/spark/built_in_lineage.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      // For each matched and not-matched outputs `ExpressionDependencyWithDelegate` is created
+      // This means for output expression id `attr.exprId.id`, `expr` node needs to be examined to 
+      // detect input expression ids. 
+      output.zipWithIndex.foreach {
+        case (attr: Attribute, index: Int) =>
+          notMatchedOutputs
+            .toStream
+            .filter(exprs => exprs.size > index)
+            .map(exprs => exprs(index))
+            .foreach(expr => deps += ExpressionDependencyWithDelegate(OlExprId(attr.exprId.id), expr))
```
Comment (mobuchowski) [inline docs/integrations/spark/built_in_lineage.md:194]: Again, there's no explanation of the delegate process.

**Review by mobuchowski — CHANGES_REQUESTED** [review]

**Comment by harels** [comment]
@pawel-big-lebowski should we take this out of draft and review since we merged #2272?

**Review by mobuchowski — COMMENTED** [review]
I like the doc, but I think it's not relevant to the users of the Spark integration. 

Rather, it's useful for people developing spark extensions, so it would be better to put it in `docs/development/developing/spark`

File: docs/development/developing/spark/built_in_lineage.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+* `io.openlineage.spark.builtin.scala.v1.InputLineageNode` with `getInputs` method,
+* `io.openlineage.spark.builtin.scala.v1.OutputLineageNode` with `getOutputs` method.
+
+They return list of `InputDatasetWithFacets` and `OutputDatasetWithFacets` respectively. Each trait has methods
+to expose dataset facets as well facets that relate to particular dataset only in the context of
+current run, like amount of bytes read from a certain dataset.
+
+### When extracting dataset name and namespace is non-trivial
+
+In the section above, the straightforward approach was used to identify dataset's `namespace` as `name`,
```
[the remark this answers is not in the document]
Comment (pawel-big-lebowski) [reply docs/development/developing/spark/built_in_lineage.md]: thanks @ddebowczyk92. I rephrased this,

**Comment by pawel-big-lebowski** [comment]
> I like the doc, but I think it's not relevant to the users of the Spark integration.
> 
> Rather, it's useful for people developing spark extensions, so it would be better to put it in `docs/development/developing/spark`

Moved whole page to development section as suggested.

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**